### PR TITLE
Add logging for alarms backwards compatability

### PR DESCRIPTION
**Pull request by johnhuangclever**
We have alarms that stopped receiving data because the kayvee log format changed:
https://app.signalfx.com/#/detector/CYi8i6PAcAM/edit?startTimeUTC=1456819200000&endTimeUTC=1459494000000

This PR will add the gearman log lines back through a legacy logger. We can remove this legacy logger once all workers are moved onto gearcmd and then alarms are switched over.

Verified through tests that the logging happens and the format is correct. Log lines generated:
```
{"function":"name","level":"info","source":"gearcmd","title":"SUCCESS","type":"counter"}
{"function":"name","level":"info","source":"gearman","title":"success","type":"counter"}
```
```
{"function":"name","level":"info","source":"gearcmd","title":"FAILURE","type":"counter"}
{"function":"name","level":"info","source":"gearman","title":"failure","type":"counter"}
```

### CODE REVIEW

**Comment by natebrennand** [comment]
Search for `kayvee.FormatLog` in the diff on https://github.com/Clever/gearcmd/pull/38/files
I think might there are a few other places we should "legacy log", I found at least one:

- [`gearman`:`failure`](https://github.com/Clever/gearcmd/pull/38/files#diff-a5d419278794525d6af4eb0055b6c5d3L74) (now is `gearcmd`:`FAILURE`)

**Comment by johnhuangclever** [comment]
Looks like the `gearman:failure` is the last one with a `gearman` source. Thanks for the heads up!

File: gearcmd/worker.go [hunk 1]
```diff
@@ -32,6 +32,10 @@ type TaskConfig struct {
 
 var (
 	lg = logger.New("gearcmd")
+	// legacy logger used to maintain existing alerts.
+	// Once all workers are migrated over to using gearcmd, and the alarms are switched over,
```
Comment (natebrennand) [inline gearcmd/worker.go]: can we note that it is gearcmd `0.5.0` or greater?

**Comment by natebrennand** [comment]
:ship: it after the last comment nit

**Comment by nathanleiby** [comment]
Created a JIRA ticket to explore this issue from infra perspective -- could we normalize logs (e.g. auto all lowercase) better? would it reduce errors like this? 

https://clever.atlassian.net/browse/INFRA-1610

cc: @azylman

**Comment by johnhuangclever** [comment]
NathanL, the source also changed from `gearman` to `gearcmd`. I'm not sure how exactly the logging pipeline handles case-sensitivity, but that was another cause of the logging mismatch.

**Comment by nathanleiby** [comment]
@johnhuangclever ah, wouldn't be able to normalize a change like that :) thanks for clarifying!

**Pull request merged**